### PR TITLE
Matching and ranking with multi-valued fields. 

### PR DESCRIPTION
**Pull request by jobergum**
testing changes:

- enable PR testing - the test URL of vespa cloud getting started should be moved as it will fail as secrets are not available for PR testing. 
- To actually test PR builds one needs to use local paths instead of pulling from what is published 

new document demonstrates searching in weightedset set strings. 

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: en/searching-multi-valued-fields.md [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+normalization and language dependent [stemming](linguistics.html#stemming).
+- String fields which shares the same [match](reference/schema-reference.html#match) 
+and linguistic processing settings can be combined using [fieldsets](reference/schema-reference.html#fieldset). 
+
+At query time, we can take the user query and translate it into a valid Vespa query request which implements
+our matching and retrieval strategy over the designed document schema.
+
+### Ranking 
+
+The documents which matches the query and is retrieved by the query is scored using a ranking model.
```
Comment (kkraune) [inline en/searching-multi-valued-fields.md:37]:
```suggestion
The documents which match the query and are retrieved by the query are scored using a ranking model. 
```

Comment (kkraune) [reply en/searching-multi-valued-fields.md:37]: or easier rewrite to "The document ..."

File: en/searching-multi-valued-fields.md [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+
+But what about `black sky`:
+
+<pre data-test="exec" data-test-assert-contains='"totalCount": 1'>
+$ vespa query 'yql=select * from photos where userQuery()' \
+ 'query=black sky' 'type=all' 'default-index=tags'
+</pre>
+
+Also matches our document. This is an example of cross-element matching. With weightedset
+usig `indexing:index` with `match:text` multi term queries match across elements.
```
Comment (kkraune) [inline en/searching-multi-valued-fields.md:289]:
```suggestion
using `indexing:index` with `match:text` multi term queries match across elements.
```

File: en/searching-multi-valued-fields.md [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+to match the set element exactly. `clear` matches `clear sky` and `sky` will match `clear sky`.  
+
+But what about `black sky`:
+
+<pre data-test="exec" data-test-assert-contains='"totalCount": 1'>
+$ vespa query 'yql=select * from photos where userQuery()' \
+ 'query=black sky' 'type=all' 'default-index=tags'
+</pre>
+
+Also matches our document. This is an example of cross-element matching. With weightedset
```
Comment (kkraune) [inline en/searching-multi-valued-fields.md:288]: consider linking to https://docs.vespa.ai/en/reference/query-language-reference.html#sameelement here, with a short note?

**Review by kkraune — APPROVED** [review]
This is a great guide! We can simplify guides/ docs in other places linking to this one instead, too, in later PRs. Thanks!

File: en/searching-multi-valued-fields.md (hunk 3 above)
Comment (jobergum) [reply en/searching-multi-valued-fields.md:288]: Problem is that sameElement does not offer any ranking features :/

Comment (kkraune) [reply en/searching-multi-valued-fields.md:288]: right. I was thinking of just the matching part, and this is a frequent support question. I'll look at the doc again and see if there is a better place to just mention this

Comment (jobergum) [reply en/searching-multi-valued-fields.md:288]: sameElement is great, but only supported for the complex array of struct and map, which per today, is only supported for attribute fields, and has no rank feature support.

**Pull request merged**